apps/web/src/pages/Communities/IndexCommunityPage/components/CommunityTable/CommunityTable.tsx

```diff
@@ -0,0 +1,200 @@
+  const intl = useIntl();
+  const paths = useRoutes();
+
+  const { pathname, search, hash } = useLocation();
+  const currentUrl = `${pathname}${search}${hash}`;
+
+  const columns = [
+    columnHelper.accessor(
+      (community) => getLocalizedName(community.name, intl),
+      {
+        id: "communityName",
+        sortingFn: normalizedText,
+        header: intl.formatMessage(commonMessages.name),
+        cell: ({ row: { original: community }, getValue }) =>
+          viewCell(
+            paths.communityView(community.id),
+            getValue(),
+            intl,
+            currentUrl,
+          ),
+        meta: {
+          isRowTitle: true,
+        },
+      },
+    ),
+    columnHelper.accessor(
+      (community) => myRolesAccessor(community.id, myRolesAndTeams, intl),
+      {
+        id: "myRoles",
+        sortingFn: normalizedText,
+        header: intl.formatMessage({
+          defaultMessage: "My roles",
+          id: "iEIzSc",
+          description: "Label displayed for the table column header My roles",
+        }),
+        cell: ({ row: { original: community } }) =>
+          myRolesCell(community.id, myRolesAndTeams, intl),
+      },
+    ),
+  ] as ColumnDef<CommunityTableCommunityFragmentType>[];
+
+  const data = communitiesQuery
+    .map((abc) => getFragment(CommunityTable_CommunityFragment, abc))
+    .filter(notEmpty);
+
+  const { roleAssignments } = useAuthorization();
+  const canCreateMembers = checkRole(
+    [ROLE_NAME.PlatformAdmin],
+    roleAssignments,
+  );
+
+  return (
+    <Table<CommunityTableCommunityFragmentType>
+      caption={title}
+      data={data}
+      columns={columns}
+      sort={{
+        internal: true,
+        initialState: [{ id: "communityName", desc: true }],
+      }}
+      pagination={{
+        internal: true,
+        total: data.length,
+        pageSizes: [10, 20, 50],
+      }}
+      search={{
+        internal: true,
+        label: intl.formatMessage({
+          defaultMessage: "Search communities",
+          id: "3hqdVV",
+          description: "Label for the communities table search input",
+        }),
+      }}
+      add={
+        canCreateMembers
+          ? {
+              linkProps: {
+                href: paths.communityCreate(),
+                label: intl.formatMessage({
+                  defaultMessage: "Create community",
+                  id: "lhLfd7",
+                  description: "Text to create a community",
+                }),
+                from: currentUrl,
+              },
+            }
+          : undefined
+      }
+      nullMessage={
+        canCreateMembers
+          ? {
+              description: intl.formatMessage({
+                defaultMessage:
+                  'Use the "Create community" button to get started.',
+                id: "Cu+CH3",
+                description: "Instructions for adding a community item",
+              }),
+            }
+          : undefined
+      }
```

Comment:
This is optional (especially since I think I advised this approach :sweat_smile: ) but we could also roll these up into a single check:

```tsx
{...(cancreateMember && {
  add: { ... },
   nullMessage: { ... }
})}
```

Reply:
e3a6d1ea8f80c16702bfa3e5120e56e70f1affd1